lib/spack/docs/packaging_guide.rst

```diff
@@ -2071,19 +2150,15 @@ The parameters are as follows:
     targets into.  It acts like a string, but it's actually its own
     special type, :py:class:`Prefix <spack.util.prefix.Prefix>`.
 
-``spec`` and ``prefix`` are passed to ``install`` for convenience.
-``spec`` is also available as an attribute on the package
-(``self.spec``), and ``prefix`` is actually an attribute of ``spec``
-(``spec.prefix``).
+The arguments ``spec`` and ``prefix`` are passed only for convenience, as they always
+correspond to ``self.spec`` and ``self.spec.prefix`` respectively.
```

Comment:
After thinking about it, I still disagree.  See  @adamjstewart comment below: "Is there any difference between prefix, self.prefix and self.spec.prefix?"  The fact remains that these arguments we're proposing to dependency-inject are ALREADY being passed to the methods, under self.  Dependency injection adds *no functional gains*.  For methods that want to use (say) the argument `spec`, there is *already* a way to make that clear, using plain vanilla programming:
```
def mymethod(self):
    spec = self.spec
    ...
```